src/main/java/me/xmrvizzy/skyblocker/skyblock/item/PriceInfoTooltip.java

```diff
@@ -20,105 +22,130 @@
+                .map(Text::getString).toList();
 
         try {
-            if (SkyblockerConfig.get().general.itemTooltip.enableNPCPrice
-                    && !listString.contains("NPC Price") && npcPricesJson != null && npcPricesJson.has(name)) {
-
-                lines.add(new LiteralText(String.format("%-21s", "NPC Price:"))
-                        .formatted(Formatting.YELLOW)
-                        .append(getCoinsMessage(npcPricesJson.get(name).getAsDouble(), count)));
+            if (SkyblockerConfig.get().general.itemTooltip.enableNPCPrice && !listString.contains("NPC Price")) {
+                if (npcPricesJson.has(name)) {
+                    lines.add(new LiteralText(String.format("%-21s", "NPC Price:"))
+                            .formatted(Formatting.YELLOW)
+                            .append(getCoinsMessage(npcPricesJson.get(name).getAsDouble(), count)));
+                }
             }
 
-            if ((!listString.contains("Avg. BIN Price") && avgPricesJson != null && avgPricesJson.has(name))
-                    || (!listString.contains("Lowest BIN Price") && lowestPricesJson != null && lowestPricesJson.has(name))) {
+            if ((!listString.contains("Avg. BIN Price") && (threeDayAvgPricesJson.has(name) || oneDayAvgPricesJson.has(name)))
+                    || (!listString.contains("Lowest BIN Price") && lowestPricesJson.has(name))) {
 
-                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN) {
+                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN && lowestPricesJson.has(name)) {
                     lines.add(new LiteralText(String.format("%-19s", "Lowest BIN Price:"))
                             .formatted(Formatting.GOLD)
                             .append(getCoinsMessage(lowestPricesJson.get(name).getAsDouble(), count)));
                 }
 
-                // Change format from Skytils to Moulberry's for Avg. BIN
                 if (name.contains("PET-")) {
                     name = name.replace("PET-", "")
-                            .replace("0", "COMMON")
-                            .replace("1", "UNCOMMON")
-                            .replace("2", "RARE")
-                            .replace("3", "EPIC")
-                            .replace("4", "LEGENDARY")
-                            .replace("5", "MYTHIC")
+                            .replace("COMMON", "0")
+                            .replace("UNCOMMON", "1")
+                            .replace("RARE", "2")
+                            .replace("EPIC", "3")
+                            .replace("LEGENDARY", "4")
+                            .replace("MYTHIC", "5")
                             .replace("-", ";");
                 } else if (name.contains("ENCHANTED_BOOK-")) {
                     name = name.replace("ENCHANTED_BOOK-", "").replace("-", ";");
+                } else if (name.contains("POTION-")) {
+                    //name = "POTION";
```

Comment:
Commented out code. Is current behavior correct or should it be replaced with the code in the comment?
Same problem below.

Reply:
reason is "remove: avg price tag from runes and potions because there is no reliable data in api" i will use comment line for explanantation.